news/5333-remove-24.7.x-deprecations

```diff
@@ -0,0 +1,95 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+* Remove `conda_build.conda_interface.Completer`. (#5333)
+* Remove `conda_build.conda_interface.CondaSession`. Use `conda.gateways.connection.session.CondaSession` instead. (#5333)
+* Remove `conda_build.conda_interface.InstalledPackages`. (#5333)
+* Remove `conda_build.conda_interface.NoPackagesFound`. Use `conda.exceptions.ResolvePackageNotFound` instead. (#5333)
+* Remove `conda_build.conda_interface.Unsatisfiable`. Use `conda.exceptions.UnsatisfiableError` instead. (#5333)
+* Remove `conda_build.conda_interface.symlink_conda`. (#5333)
+* Remove `conda_build.conda_interface.ArgumentParser`. Use `conda.cli.conda_argparse.ArgumentParser` instead. (#5333)
+* Remove `conda_build.conda_interface.add_parser_channels`. Use `conda.cli.helpers.add_parser_channels` instead. (#5333)
+* Remove `conda_build.conda_interface.add_parser_prefix`. Use `conda.cli.helpers.add_parser_prefix` instead. (#5333)
```

Comment:
Should we link out to a doc, summarize or streamline the list a bit?

An example of streamlining:

```markdown
### Deprecations

* Remove `conda_build.conda_interface.*` component. (#5333):
  * `Completer`
  * ...
```

Reply:
That's a great idea!